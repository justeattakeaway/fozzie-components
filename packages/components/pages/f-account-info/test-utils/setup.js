import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import { VueI18n } from '@justeat/f-globalisation';
import tenantConfigs from '../src/tenants';

export const localVue = createLocalVue();
localVue.use(VueI18n);
localVue.use(Vuex);

export const i18n = {
    locale: 'en-GB',
    messages: {
        'en-GB': tenantConfigs['en-GB'].messages
    }
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const consumerDetailsGetResponse = {
    data: {
        ConsumerId: 1050450174,
        Id: '8It2lEaJjG82nJtxO00Iab0CoF0=',
        CreatedDate: '2019-04-16T17:25:00.000Z',
        EmailAddress: 'satchin.moriyoka@example.com',
        FirstName: 'Satchin',
        LastName: 'Moriyoka',
        PhoneNumber: '+441234567890',
        DateOfBirth: '1914-04-16T17:25:00.000Z',
        TermsAndConditionsId: 0,
        WantsNewsletter: true,
        AccountType: 'Default',
        ConsumerStatus: 'ReturnConsumer'
    }
};

const consumerAddressGetResponse = {
    AddressId: 1050450174,
    City: 'Bristol',
    ZipCode: 'BS1 4DJ',
    AddressName: 'Office',
    IsDefault: true, // Key field
    Line1: 'Just Eat',
    Line2: 'Broad Quay House'
};

export const consumerAddressesGetResponse = {
    data: [{
        ...consumerAddressGetResponse
    },
    {
        AddressId: 1050450175,
        City: 'Bristol',
        ZipCode: 'BS1 9DJ',
        AddressName: 'Office',
        IsDefault: false,
        Line1: 'Takeaway',
        Line2: 'Broad Quay House'
    }]
};

export const consumerDetailsMappedModel = {
    id: consumerDetailsGetResponse.data.ConsumerId,
    firstName: consumerDetailsGetResponse.data.FirstName,
    lastName: consumerDetailsGetResponse.data.LastName,
    phoneNumber: consumerDetailsGetResponse.data.PhoneNumber,
    emailAddress: consumerDetailsGetResponse.data.EmailAddress
};

export const consumerAddressMappedModel = {
    line1: consumerAddressGetResponse.Line1,
    line2: consumerAddressGetResponse.Line2,
    line3: consumerAddressGetResponse.Line3,
    locality: consumerAddressGetResponse.City,
    postcode: consumerAddressGetResponse.ZipCode
};

export const consumerViewModel = {
    ...consumerDetailsMappedModel,
    ...consumerAddressMappedModel
};

export const consumerStateModel = {
    consumer: {
        ...consumerDetailsMappedModel,
        ...consumerAddressMappedModel
    }
};
