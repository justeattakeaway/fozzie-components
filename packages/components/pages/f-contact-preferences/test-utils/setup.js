export const contactPreferencesGetResponse = {
    data: {
        Preferences: [{
            DisplayName: 'Order status',
            Sort: 1,
            Key: 'orderstatus',
            Push: false,
            Email: true,
            Sms: true
        }, {
            DisplayName: 'Review meal',
            Sort: 2,
            Key: 'reviewmeal',
            Push: false,
            Email: false,
            Sms: false
        }, {
            DisplayName: 'News & offers',
            Sort: 3,
            Key: 'news',
            Push: false,
            Email: false,
            Sms: false
        }
        ],
        PreferenceVersionViewed: 0
    }
};

export const contactPreferencesViewModel = {
    preferenceVersionViewed: 0,
    preferences: [{
        displayName: 'Order status',
        isEmailEnabled: false,
        emailValue: true,
        key: 'orderstatus',
        pushEnabled: false,
        pushValue: false,
        isSmsEnabled: false,
        smsValue: true,
        sort: 1,
        visible: true
    }, {
        displayName: 'Review meal',
        isEmailEnabled: true,
        emailValue: false,
        key: 'reviewmeal',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 2,
        visible: false
    }, {
        displayName: 'News & offers',
        isEmailEnabled: true,
        emailValue: false,
        key: 'news',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 3,
        visible: true
    }]
};

export const filteredContactPreferencesModel = [{
    displayName: 'Order status',
    isEmailEnabled: false,
    emailValue: true,
    key: 'orderstatus',
    pushEnabled: false,
    pushValue: false,
    isSmsEnabled: false,
    smsValue: true,
    sort: 1,
    visible: true
}, {
    displayName: 'News & offers',
    isEmailEnabled: true,
    emailValue: false,
    key: 'news',
    pushEnabled: true,
    pushValue: false,
    isSmsEnabled: true,
    smsValue: false,
    sort: 3,
    visible: true
}];

export const contactPreferencesUpdateModel = {
    Preferences: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: false,
        Sms: false
    }],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const contactPreferencesUpdateBody = {
    Preference: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: true,
        Sms: false
    }
    ],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};
