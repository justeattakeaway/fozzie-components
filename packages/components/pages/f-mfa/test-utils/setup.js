export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

// eslint-disable-next-line camelcase
export const postBody = { mfa_token: 'some_mfa_token', otp: 'ABC123' };
