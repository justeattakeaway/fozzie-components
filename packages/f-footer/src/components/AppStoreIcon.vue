<template>
    <component :is="iconComponent" />
</template>

<script>
import {
    AppAndroidDkIcon as AndroidIconDaDk,
    AppAndroidIcon as AndroidIconEnGb,
    AppAndroidEsIcon as AndroidIconEsEs,
    AppAndroidItIcon as AndroidIconItIt,
    AppAndroidNoIcon as AndroidIconNbNo,
    AppIosDkIcon as IosIconDaDk,
    AppIosIcon as IosIconEnGb,
    AppIosEsIcon as IosIconEsEs,
    AppIosItIcon as IosIconItIt,
    AppIosNoIcon as IosIconNbNo
} from '@justeat/f-vue-icons';

export default {
    components: {
        AndroidIconDaDk,
        AndroidIconEnGb,
        AndroidIconEnIe: AndroidIconEnGb,
        AndroidIconEnAu: AndroidIconEnGb,
        AndroidIconEnNz: AndroidIconEnGb,
        AndroidIconEsEs,
        AndroidIconItIt,
        AndroidIconNbNo,
        IosIconDaDk,
        IosIconEnGb,
        IosIconEnIe: IosIconEnGb,
        IosIconEnAu: IosIconEnGb,
        IosIconEnNz: IosIconEnGb,
        IosIconEsEs,
        IosIconItIt,
        IosIconNbNo
    },
    props: {
        name: {
            type: String,
            required: true
        },
        locale: {
            type: String,
            required: false,
            default: 'en-GB'
        }
    },
    computed: {
        iconComponent () {
            return `${this.name}-icon-${this.locale.toLowerCase()}`;
        }
    }
};
</script>
