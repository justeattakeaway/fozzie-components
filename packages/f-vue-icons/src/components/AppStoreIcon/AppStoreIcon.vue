<template>
    <component :is="iconComponent" />
</template>

<script>
import AndroidIconDaDk from '@justeat/f-icons/src/img/icons/appstore/android.dk.svg';
import AndroidIconEnGb from '@justeat/f-icons/src/img/icons/appstore/android.svg';
import AndroidIconEsEs from '@justeat/f-icons/src/img/icons/appstore/android.es.svg';
import AndroidIconItIt from '@justeat/f-icons/src/img/icons/appstore/android.it.svg';
import AndroidIconNbNo from '@justeat/f-icons/src/img/icons/appstore/android.no.svg';
import IosIconDaDk from '@justeat/f-icons/src/img/icons/appstore/ios.dk.svg';
import IosIconEnGb from '@justeat/f-icons/src/img/icons/appstore/ios.svg';
import IosIconEsEs from '@justeat/f-icons/src/img/icons/appstore/ios.es.svg';
import IosIconItIt from '@justeat/f-icons/src/img/icons/appstore/ios.it.svg';
import IosIconNbNo from '@justeat/f-icons/src/img/icons/appstore/ios.no.svg';
import MicrosoftIconDaDk from '@justeat/f-icons/src/img/icons/appstore/microsoft.dk.svg';

export default {
    components: {
        AndroidIconDaDk,
        AndroidIconEnGb,
        AndroidIconEnIe: AndroidIconEnGb,
        AndroidIconEnAu: AndroidIconEnGb,
        AndroidIconEnNz: AndroidIconEnGb,
        AndroidIconEsEs,
        AndroidIconItIt,
        AndroidIconNbNo,
        IosIconDaDk,
        IosIconEnGb,
        IosIconEnIe: IosIconEnGb,
        IosIconEnAu: IosIconEnGb,
        IosIconEnNz: IosIconEnGb,
        IosIconEsEs,
        IosIconItIt,
        IosIconNbNo,
        MicrosoftIconDaDk
    },
    props: {
        name: {
            type: String,
            required: true
        },
        locale: {
            type: String,
            required: false,
            default: 'en-GB'
        }
    },
    computed: {
        iconComponent () {
            return `${this.name}-icon-${this.locale.toLowerCase()}`;
        }
    }
};
</script>
