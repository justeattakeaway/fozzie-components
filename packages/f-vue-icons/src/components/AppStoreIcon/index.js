import AppStoreIcon from './AppStoreIcon.vue';

const install = Vue => {
    Vue.component(AppStoreIcon.name, AppStoreIcon);
};

export default install;
