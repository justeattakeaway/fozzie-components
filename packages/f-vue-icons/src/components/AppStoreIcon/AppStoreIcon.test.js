import { shallowMount } from '@vue/test-utils';
import AppStoreIcon from './AppStoreIcon.vue';

let propsData = { name: 'ios' };

describe('AppStoreIcon', () => {
    it('should be defined', () => {
        const wrapper = shallowMount(AppStoreIcon, {
            propsData
        });
        expect(wrapper).toBeDefined();
    });

    it('default component markup is rendered', () => {
        // Arrange & Act
        const wrapper = shallowMount(AppStoreIcon, {
            propsData
        });

        // Assert
        expect(wrapper).toMatchSnapshot();
    });

    it('Ios component for GB should be rendered if ios name is passed as a prop and locale is default', () => {
        // Arrange
        const wrapper = shallowMount(AppStoreIcon, {
            propsData
        });

        // Act
        const result = wrapper.vm.iconComponent;

        // Assert
        expect(result).toBe('ios-icon-en-gb');
    });

    it('Ios component for DK should be rendered if ios name is passed as a prop and da-DK locale is passed as a prop', () => {
        // Arrange
        propsData = { name: 'ios', locale: 'da-DK' };
        const wrapper = shallowMount(AppStoreIcon, {
            propsData
        });

        // Act
        const result = wrapper.vm.iconComponent;

        // Assert
        expect(result).toBe('ios-icon-da-dk');
    });
});
