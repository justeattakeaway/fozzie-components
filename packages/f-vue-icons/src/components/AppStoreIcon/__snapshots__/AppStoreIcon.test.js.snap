// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppStoreIcon default component markup is rendered 1`] = `<ios-icon-en-gb-stub></ios-icon-en-gb-stub>`;
